src/app/components/wrapper-components/rules-configuration/rules-configuration.component.ts

```diff
@@ -0,0 +1,220 @@
+            return configuration[name];
+        } else {
+            return defaultValue;
+        }
+    }
+
+    private getFormControl(value: ConfigDescriptionType, configurable: boolean): FormControl {
+        const formControl: FormControl = new FormControl(value);
+        if (configurable === false) {
+            formControl.disable();
+        }
+        formControl.valueChanges.subscribe(() => {
+            this.onUpdate();
+        });
+        return formControl;
+    }
+
+    public onUpdate(): void {
+        if (this.userIsCreator === false) {
+            ErrorLoggerService.logError('RulesConfiguration', 'Only creator should be able to modify rules config');
+        } else if (this.chosenConfigName !== 'Custom') {
+            ErrorLoggerService.logError('RulesConfiguration', 'Only Customifiable config should be modified!');
+        } else {
+            const rulesConfig: RulesConfig = {};
+            const config: RulesConfig = this.rulesConfigDescription.getDefaultConfig().config;
+            const parameterNames: string[] = Object.keys(config);
+            for (const parameterName of parameterNames) {
+                if (this.isValid(parameterName)) {
+                    rulesConfig[parameterName] = this.rulesConfigForm.controls[parameterName].value;
+                } else {
+                    // This informs the parent component that an invalid update has been done
+                    this.updateCallback.emit(MGPOptional.empty());
+                    return; // In order not to send update when form is invalid
+                }
+            }
+            this.setConfigDemo(rulesConfig);
+            this.updateCallback.emit(MGPOptional.of(rulesConfig));
+        }
+    }
+
+    public isNumber(field: string): boolean {
+        const config: RulesConfig = this.rulesConfigDescription.getDefaultConfig().config;
+        const value: ConfigDescriptionType = config[field];
+        return typeof value === 'number';
+    }
+
+    public isBoolean(field: string): boolean {
+        const config: RulesConfig = this.rulesConfigDescription.getDefaultConfig().config;
+        const value: ConfigDescriptionType = config[field];
+        return typeof value === 'boolean';
+    }
+
+    public isValid(field: string): boolean {
+        const config: RulesConfig = this.rulesConfigDescription.getDefaultConfig().config;
+        const value: ConfigDescriptionType = config[field];
+        if (typeof value === 'number') {
+            const fieldValue: number = this.rulesConfigForm.controls[field].value;
+            const validity: MGPValidation = this.rulesConfigDescription.getValidator(field)(fieldValue);
+            return validity.isSuccess();
+        } else {
+            Utils.expectToBe(typeof value, 'boolean');
+            // Angular make those controls invalid when they are boolean, not sure why
+            return true; // So we return false cause they cannot be invalid, checked or not, its always valid
```

Comment:
"we return false": non non, we return true !
et "because they are always valid"